pkg/inject/init_proxy.go

```diff
@@ -102,6 +96,14 @@ func (m *initProxyMutator) mutate(pod *corev1.Pod) error {
 	if err != nil {
 		return err
 	}
+
+	// add resource requests and limits
+	container.Resources, err = sidecarResources(m.mutatorConfig.cpuRequests, m.mutatorConfig.memoryRequests,
```

Comment:
Why are we not checking for pod annotations here for resource limits?

Reply:
init container does not really depend on your application datapath load. Init container here will run as one shot, do it's job and finish execution. Whereas envoy or xray may need higher or lower resource limits depending on your application as they come in the datapath

Reply:
There might be a bit of user experience aspect that users would need to know how to enable limits for init containers. We will document it